src/SourceGenerators/InstrumentationDefinitionsGenerator.cs

```diff
@@ -0,0 +1,280 @@
+
+        sb.AppendLine(@"        return nativeCallTargetDefinitions.ToArray();
+    }
+}");
+
+        return sb.ToString();
+    }
+
+    private static void GenerateInstrumentationForSignal(IReadOnlyCollection<IntegrationToGenerate> integrations, StringBuilder sb, string signalHeader, string conditionPrefix)
+    {
+        if (integrations.Count > 0)
+        {
+            sb.AppendLine(signalHeader)
+                .AppendLine("{");
+            GenerateIntegrations(integrations, sb, conditionPrefix);
+
+            sb.AppendLine("        }")
+                .AppendLine();
+        }
+    }
+
+    private static void GenerateIntegrations(IReadOnlyCollection<IntegrationToGenerate> integrations, StringBuilder sb, string conditionPrefix)
+    {
+        var gropedByIntegrationName = integrations.GroupBy(x => x.IntegrationName);
+
+        foreach (var group in gropedByIntegrationName)
+        {
+            sb.Append("            // ");
+            sb.AppendLine(group.Key);
+            sb.AppendFormat(
+                @"            if ({0}.{1}))
+            {{",
+                conditionPrefix,
+                group.Key);
+            sb.AppendLine();
+
+            foreach (var integration in group)
+            {
+                sb.AppendFormat(
+                    "                nativeCallTargetDefinitions.Add(new(\"{0}\", \"{1}\", \"{2}\", new[] {{{3}}}, {4}, {5}, {6}, {7}, {8}, {9}, AssemblyFullName, \"{10}\"));",
+                    integration.TargetAssembly,
+                    integration.TargetType,
+                    integration.TargetMethod,
+                    string.Join(", ", integration.TargetSignatureTypes!.Select(x => $"\"{x}\"")),
+                    integration.TargetMinimumMajor,
+                    integration.TargetMinimumMinor,
+                    integration.TargetMinimumPatch,
+                    integration.TargetMaximumMajor,
+                    integration.TargetMaximumMinor,
+                    integration.TargetMaximumPatch,
+                    integration.IntegrationType);
+                sb.AppendLine();
+            }
+
+            sb.AppendLine("            }");
+            sb.AppendLine();
+        }
+    }
+
+    private static bool IsAttributedClass(SyntaxNode node)
+    {
+        return node is ClassDeclarationSyntax { AttributeLists.Count: > 0 };
+    }
+
+    private static ClassDeclarationSyntax? GetClassesMarkedByInstrumentMethodAttribute(GeneratorSyntaxContext context)
+    {
+        var classDeclarationSyntax = (ClassDeclarationSyntax)context.Node;
+
+        foreach (var attributeListSyntax in classDeclarationSyntax.AttributeLists)
+        {
+            foreach (var attributeSyntax in attributeListSyntax.Attributes)
+            {
+                if (attributeSyntax.Name.ToString() is "InstrumentMethod" or "InstrumentMethodAttribute" or "GraphQLExecuteAsync" or "AdoNetTargetSignature")
```

Comment:
You can remove ` or "GraphQLExecuteAsync" or "AdoNetTargetSignature"` from the condition.

You need only one of `"InstrumentMethod" or "InstrumentMethodAttribute"`, no?

Reply:
f02feab0ac6f361fbf3b6babe83aa2e3aa99abbf